services-custom/dynamodb-enhanced/README.md

```diff
@@ -291,32 +296,205 @@ Using composition:
+```
+
+### Mark attributes that should participate in mapping to DDB
+Any attribute in a class marked with @DynamoDbBean will be included in the mapping as long
+as it has a getter and a setter, according to the Java Bean standard. It's also
+possible to explicitly mark attributes to participate in the mapping to DDB, with a custom name,
+using the @DynamoDbAttribute annotation:
+```java
+private String internalKey;
+
+@DynamoDbAttribute("renamedInternalKey")
+public String getInternalKey() { return this.internalKey; }
+public void setInternalKey(String internalKey) { return this.internalKey = internalKey;}
+```
+
+### Flat map attributes from another class
+If the attributes for your table record are spread across several
+different Java objects, either through inheritance or composition, the
+BeanTableSchema provides methods to flat map
+those attributes and rolling them up into a single schema.
+
+To accomplish this using inheritance, the only requirement is that
+both classes are annotated as a DynamoDb bean:-
+```java
+@DynamoDbBean
+public class Customer extends GenericRecord {
+    private String name;
+    private GenericRecord record;
+
+    public String getName() { return this.name; }
+    public void setName(String name) { this.name = name;}
+
+    public String getRecord() { return this.record; }
+    public void setRecord(String record) { this.record = record;}
+}
+
+@DynamoDbBean
+public abstract class GenericRecord {
+    private String id;
+    private String createdDate;
+
+    public String getId() { return this.id; }
+    public void setId(String id) { this.id = id;}
+
+    public String getCreatedDate() { return this.createdDate; }
+    public void setCreatedDate(String createdDate) { this.createdDate = createdDate;}
+}
+
+```
+
+Using composition, the @DynamoDbFlatten annotation flat maps the composite class:
+```java
+@DynamoDbBean
+public class Customer {
+    private String name;
+    private GenericRecord record;
+
+    public String getName() { return this.name; }
+    public void setName(String name) { this.name = name;}
+
+    @DynamoDbFlatten(dynamoDbBeanClass = GenericRecord.class)
+    public String getRecord() { return this.record; }
+    public void setRecord(String record) { this.record = record;}
+}
+
+@DynamoDbBean
+public class GenericRecord {
+    private String id;
+    private String createdDate;
+
+    public String getId() { return this.id; }
+    public void setId(String id) { this.id = id;}
+
+    public String getCreatedDate() { return this.createdDate; }
+    public void setCreatedDate(String createdDate) { this.createdDate = createdDate;}
+}
+```
+You can flatten as many different eligible classes as you like using the flatten annotation.
+The only constraints are that attributes must not have the same name when they are being rolled
+together, and there must never be more than one partition key, sort key or table name.
+
+### Control how attributes are converted
```

Comment:
Same comment as before, I think we should organize by use-case and then a description of how to do it in the BeanTableSchema and then the StaticTableSchema